packages/bot/src/transformers/entitlement.ts

```diff
@@ -14,6 +14,7 @@ export function transformEntitlement(bot: Bot, payload: DiscordEntitlement): Ent
   if (props.deleted && payload.deleted) entitlement.deleted = payload.deleted
   if (props.startsAt && payload.starts_at) entitlement.startsAt = Date.parse(payload.starts_at)
   if (props.endsAt && payload.ends_at) entitlement.endsAt = Date.parse(payload.ends_at)
+  if (props.consumed && payload.consumed) entitlement.consumed = payload.consumed
```

Comment:
```suggestion
  if (props.consumed && payload.consumed !== undefined) entitlement.consumed = payload.consumed
```
Actually... why do we even need to check if the payload contains something?

Reply:
idr exactly the reason but we do so...

Reply:
I think I figured out the reason, sometimes we pass partial stuff to the transformers and abused `ts-expect-error` as well as pray that we handled optional payload properly in the transformer.

Reply:
This is the reaction for #3613, i have opened it while doing stuff for #3618 and started thinking that the abuse of ts-expect-error is getting out of hand

Reply:
Alright got it thank you! ❤️